compiler/rustc_middle/src/metadata.rs

```diff
@@ -43,4 +43,6 @@ pub struct ModChild {
     /// Reexport chain linking this module child to its original reexported item.
     /// Empty if the module child is a proper item.
     pub reexport_chain: SmallVec<[Reexport; 2]>,
+    /// Is this an ambiguous binding?
+    pub ambiguity: bool,
```

Comment:
The `module_children` query is used in different situations (mostly for diagnostics), but in all of them except the one in `build_reduced_graph` it should not include ambiguities.

I think ambiguous children need to be stored to a separate table next to regular `module_children` tables, and then only additionally read in `build_reduced_graph_external`.